portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,9 +70,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String title = getParameter(request, "comment-input", "");
         long timestamp = System.currentTimeMillis();
-
+        UserService userService = UserServiceFactory.getUserService();
+        if (!userService.isUserLoggedIn()) {
+            response.sendRedirect("/shoutbox");
+            return;
+        }
+        String email = userService.getCurrentUser().getEmail();
+        
         Entity taskEntity = new Entity("Task");
-
+        taskEntity.setProperty("id", email);
```

Comment:
Should this be "author" instead of "id"?

Reply:
Just fixed that will add another commit!